src/buildConfig.js

```diff
@@ -44,7 +46,7 @@ export default function (options) {
   const configBasedir = options.configBasedir || false
 
   if (rawConfig) {
-    const configDir = configBasedir || process.cwd()
+    const configDir = configBasedir || (options.filepath ? path.dirname(options.filepath) : process.cwd())
```

Comment:
Where is `options.filepath` coming from? Why isn't it in camelCase?

[The Documentation](http://stylelint.io/user-guide/node-api/#configfile) already says the path is based on the `cwd()`, but this PR doesn't update it to account for this changed behavior.

This completely ignores the `codeFilename` value from the Node API.